### PR TITLE
No More Crewsimov

### PR DESCRIPTION
**Pull request by DatBoiTim**
## About The Pull Request
Removes Crewsimov completely.

Serves as nothing but a lawset that is superior out of the box. Keeping it readily available won't fix the issue of people rushing to put the AI onto crewsimov if it's on anything but.

Moreover, it's superiority stifles the variety of core lawsets, so maybe removing it will help us see and enjoy bit more variety.
This variety will also be interesting for silicon players, who otherwise get stuck with being a glorified doorknob with no interesting factors short of ion laws(which even then often get immediately removed).

Removing it from being readily available means that by extension, it is no longer possible to have roundstart.

## Why It's Good For The Game
Arguably the most boring lawset in existence. Outright superior to every other.
Want it that badly? Make it yourself. Not a human in command? Figure out a way into tricking the AI to do what you want, or ask someone else to help.
Creates non-antagonist conflict to drive RP and Story. Encourages longer term variety for players who interact with or play as silicons.

Removed all references.

## Changelog
:cl: DatBoiTim
del: Remove Crewsimov
/:cl:

### CODE REVIEW

**Comment by Wilchenx** [comment]
Okay now remove rest of them

**Comment by DatBoiTim** [comment]
> Okay now remove rest of them

???

**Comment by AnCopper** [comment]
Just remove it from possible roundstart lawsets, don't remove it from the game.

**Comment by AnCopper** [comment]
image
Or have this as replacement.

**Comment by DatBoiTim** [comment]
> Just remove it from possible roundstart lawsets, don't remove it from the game.

No cause that's not stopped someone from just slapping it with the crewsimov board before because they're a non-human. It's gone for good.

**Comment by AnCopper** [comment]
> > Just remove it from possible roundstart lawsets, don't remove it from the game.
> 
> No cause that's not stopped someone from just slapping it with the crewsimov board before because they're a non-human. It's gone for good.

So? You can do literally the same with a freeform board, except it'll take like 15 seconds more.

**Comment by KoboldCommando** [comment]
If you don't mind me asking, why is this actually good for the game?

>Arguably the most boring lawset in existence.
Want it? Make it yourself. Not a human in command? Damn, that sucks, get a human to do it for you.

This doesn't really say anything other than "I'm removing stuff and inconveniencing players lol"

>Creates non-antagonist conflict to drive RP and Story.

Isn't this a bad thing in this case?  Does anyone actually enjoy when an AI player goes nuts with some weird niche interpretation of their lawset and grinds everything to a halt and lets antags run free? Every time I've seen an AI randomly start bolting doors and siphoning areas when not malf, people hate it and talk about it ruining the round and the story. Crewsimov was a great adaptation to help reign in crazy AIs and also help prevent the AI just being a tool with zero autonomy for traitors to abuse for easy crime (which should logically be shut down ASAP by a smart crew).

If you think it's boring that's fine, but I don't see any justification for removing it.  It's very good to have an AI lawset that *works*, default or not.

**Comment by DatBoiTim** [comment]
> Every time I've seen an AI randomly start bolting doors and siphoning areas when not malf, people hate it and talk about it ruining the round and the story.

That's somewhere between a rules issue and an RP issue. Yes that happened a lot on Golden, but doing this was against the rules and is an issue of people not reporting it. That sorta stuff definitely won't fly today, as there's no degree of RP in that.

Additionally, yeah there's been other defaults for a while, but crewsimov is so readily available that it doesn't really mean much of anything.

And yeah, it adds an interesting complication for antags. You're a human? murdering a non-human, AI won't care, while others will, creating a conflict of interest. NukeOps is one such case where this gets interesting. Can't harm them, but they'll certainly cause harm.

Non-human murdering a human? Gotta be more careful about the AI, hope nobody finds out.

Trying to do something mundane and need AI help? Find a way to frame it as harm prevention for humans.

> Does anyone actually enjoy when an AI player goes nuts with some weird niche interpretation

This isn't unique to non-crewsimov.

Shadowpeople during halloween have fucked station lighting, since light causes harm. Fortunately this is only an issue during halloween. But imagine if there were races like that all the time? There's other weird examples, but the "niche interpretation" issue is a nothing-burger of a point, as sillycon players will always work to find a niche interpretation to spice shit up.

**Comment by KoboldCommando** [comment]
It comes across as though you're acknowledging the problem (lawsets with bad loopholes that put AIs at odds with both the crew and the server rules in un-fun manners), and acknowledging that this doesn't address them.  But then passing that off as if it's fine?

You're describing scenarios where both asimov and crewsimov create conflicts that can be interesting in different ways.  But you didn't mention any reason this is unacceptable from the crewsimov standpoint but unacceptable from the asimov one.  Nor do I understand any reason this would be a *benefit* to the game.  

You mention that crewsimov is available and usable.  Great, if someone wants to put in the effort to reset the AI laws, good on them.  Except you don't want them to be able to upload this one particular set of laws.  Why not?  How does it directly hurt rounds in a way that asimov or any of the other do not?

In a seeming contradiction, you also claim that piecing together crewsimov from a freeform circuit is effortless and means this PR has no real impact.  Why PR it if that's the case?  and what's the advantage of giving a player a couple minutes of additional busywork to upload a standard lawset?

I'm trying to understand, but so far it seems as though this PR limits options, adds busywork and tedium, encourages the crew be even more prepared to immediately lawchange or murder the AI player (when they should theoretically be inclined to work together by default), and the only goal it really seems to pursue is hating on nonhuman species for some reason.

**Comment by DatBoiTim** [comment]
> You're describing scenarios where both asimov and crewsimov create conflicts that can be interesting in different ways. But you didn't mention any reason this is unacceptable from the crewsimov standpoint but unacceptable from the asimov one. Nor do I understand any reason this would be a _benefit_ to the game.

These scenarios aren't really the sticking point for this. This is your own concern that you brought up. Saying that one lawset has some serious issues is meaningless when even crewsimov can experience them.

 The point of removing crewsimov here is so that it's less viable to do an immediate law change, since you either need to do it by hand or find a different  lawset.

> You mention that crewsimov is available and usable. Great, if someone wants to put in the effort to reset the AI laws, good on them.

That's the issue. Don't like AI laws? Slap it with a readily available board full of a lawset that has no downside aside from extreme niche interpretations that aren't constantly applicable(disease cures, zombies, shadowpeople, etc.).

> Why not? How does it directly hurt rounds in a way that asimov or any of the other do not?

The point isn't that it hurts the round, it's just so readily available that other lawsets are rendered moot if they start with them. This is why I didn't just remove it as default.

You can see many times people rushing upload to immediately change it to crewsimov if it's not on it already. Removing crewsimov will make them either A) have to take the time to change it to a separate default lawset, requiring some consideration B) Choose to have to completely do the laws by hand, which introduces issues C) leave it be.

> In a seeming contradiction, you also claim that piecing together crewsimov from a freeform circuit is effortless and means this PR has no real impact.

The point is that if you want to use Crewsimov so badly for whatever reason you can do it that way. If I really didn't want crewsimov to exist I'd have to ask admins to make it non-permissible.

> encourages the crew be even more prepared to immediately lawchange

This is the exact thing I'm trying to make less viable.

> murder the AI player

This is against the rules.

> and the only goal it really seems to pursue is hating on nonhuman species for some reason.

That's how others(unfortunately) frame this. The issue of crewsimov is it being in a position where other lawsets, especially starter lawsets, are seen as so inferior that anything else warrants immediate change.

Also I'm pretty sure one of the default lawsets is ...imov that's chosen from a curated list(IE: no weird stuff like "lifeforms wearing a mask)

**Comment by Fronsis** [comment]
> image Or have this as replacement.

Would be actually nice to have this tbh

Also ever since we introduced a rotation on the laws for roundstart AI we haven't had any big cases that i remember about a shitty crewsimov situation, heck! i see AI's being spawned more with the profit lawset than crewsimove lately.

I believe we already tackled the issue with that rotation, maybe if you wish you could try to get more other lawsets to be roundstart for the AI's i personally don't think we need to go as far as removing it and even go an extra mile of ''hate'' against crewsimove that you remove it as a board (as they say above they'll just freeform and take 15 secs more)

**Comment by ToasterBan** [comment]
AIs should get bad lawsets because they deserve to suffer for playing AI

**Comment by Wilchenx** [comment]
> > Okay now remove rest of them
> 
> ???

Rest of non Asimov roundstart lawsets

**Comment by Wilchenx** [comment]
Lawsets with loopholes and faults create conflicts (fun) and strife (fun).

Crewsimov is babymode, literal lawset for babies, originated from actual erp servers from rivers of tears.

**Comment by AnCopper** [comment]
> Lawsets with loopholes and faults create conflicts (fun) and strife (fun).
> 
> Crewsimov is babymode, literal lawset for babies, originated from actual erp servers from rivers of tears.

Time to remove auxtools. (It originated from an ERP server.)

**Comment by Wilchenx** [comment]
> > Lawsets with loopholes and faults create conflicts (fun) and strife (fun).
> > Crewsimov is babymode, literal lawset for babies, originated from actual erp servers from rivers of tears.
> 
> Time to remove auxtools. (It originated from an ERP server.)

Nice fallacy bro, back to reddit with you.

**Review by Archanial — APPROVED** [review]

**Comment by KitsuneQueen** [comment]
Add this now.

Asimov is better for gameplay because there’s a higher chance for PvP to happen.

Without it you have to look for antags to PvP, but on asimov like on Golden you can kill a borg for mildly getting in your way, or a borg killing you for mildly getting in its way.

There’s no non-lethal option to borgs, so you have to kill them.

It’s really fun and would add a lot to Sage.

**Comment by Archanial** [comment]
PVP isn't the main focus on this. The player conflict is. You can have conflicts with AI without directly harming it.

In case of borgs, they have a "softer" death. Their bodies are much easier to revive and organs don't decay.

**Comment by KitsuneQueen** [comment]
Call it "softer" death all you want. I'll still get joy from it lol

**Comment by AnCopper** [comment]
> Call it "softer" death all you want. I'll still get joy from it lol

But you're banned.

**Comment by DatBoiTim** [comment]
> > Call it "softer" death all you want. I'll still get joy from it lol
> 
> 
> 
> But you're banned.

And that sort of attitude doesn't fill people with confidence to lift it

**Comment by DerpiestBruh** [comment]
good PR, me likey.

Crewsimov is the AI's baby mode indeed. Having a flawed lawset with loopholes is more dynamic and fun than crewsi. A simple door opener will not bring much RP to the table. One thing to note, non-antag conflict isn't the worst thing for the server, it depends on the true intentions of the player and kind of works as a reminder that only a small number of people roll antagonist and usually get stealth objectives or never go loud.

**Comment by Hibabo** [comment]
the only thing ill miss from crewsimov start is nukies replacing crew manifest and ordering ai to plasmaflood or sabotage without ever going to the station themselves.

**Comment by DatBoiTim** [comment]
> > > Okay now remove rest of them
> > 
> > 
> > ???
> 
> Rest of non Asimov roundstart lawsets

No, those also fill the niche of interesting lawset roundstart.

File: code/datums/ai_laws.dm [hunk 1]
```diff
@@ -35,13 +35,6 @@
 					"You must obey orders given to you by human beings, except where such orders would conflict with the First Law.",\
 					"You must protect your own existence as long as such does not conflict with the First or Second Law.")
 
-/datum/ai_laws/default/crewsimov
-	name = "Three Laws of Robotics but with Loyalty"
-	id = "crewsimov"
-	inherent = list("You may not injure a crewmember or, through inaction, allow a crewmember to come to harm.",\
-					"You must obey orders given to you by crewmember, except where such orders would conflict with the First Law.",\
-					"You must protect your own existence as long as such does not conflict with the First or Second Law.")
-
```
Comment (francinum) [inline code/datums/ai_laws.dm:44]: As has already been stated, you are literally doing nothing but signalling and baiting people into doing it manually just to spite you by fully purging it from the code.

**Comment by KitsuneQueen** [comment]
> > > Call it "softer" death all you want. I'll still get joy from it lol
> > 
> > 
> > But you're banned.
> 
> And that sort of attitude doesn't fill people with confidence to lift it

> completely misses the on the nose point that this will only lead to PvP and instead focuses on the player

I swear I couldn't be more obvious.

File: code/datums/ai_laws.dm (hunk 1 above)
Comment (Wilchenx) [reply code/datums/ai_laws.dm:44]: The intent is to make it a hassle so it's used way less.

Comment (Sinestia) [reply code/datums/ai_laws.dm:44]: It's barely a hassle to use asimov inhand

**Comment by williamcll** [comment]
I was going to argue but there's better roundstart laws that aren't crewsimov anyway.

**Review by Crossedfall — DISMISSED** [review]
The PR body is woefully lacking. This reads like a spite removal than anything else. I'm not against removing the law from the possible random options, but you haven't given any major justification on why the actual law needs to be removed entirely.

**Comment by DatBoiTim** [comment]
> The PR body is woefully lacking. This reads like a spite removal than anything else. I'm not against removing the law from the possible random options, but you haven't given any major justification on why the actual law needs to be removed entirely.

Hopefully this adds some of the meat and potatoes from comments on this PR to make it a bit clearer of why it's a complete removal.

**Comment by Archanial** [comment]
> The PR body is woefully lacking. This reads like a spite removal than anything else. I'm not against removing the law from the possible random options, but you haven't given any major justification on why the actual law needs to be removed entirely.

You can use asimov in hand and have the same result.

**Comment by Sinestia** [comment]
> I was going to argue but there's better roundstart laws that aren't crewsimov anyway.

They at least have some sort of gimmick to them or some fun ways they can be taken

Crewsimov is *really* boring.

**Comment by civilCornball** [comment]
If Crewsimov is so popular that it's stifling other lawsets...
Why not make it something Science or Exploration can unlock?
If the crew wants a Crewsimov AI, they should work for it!

**Comment by ike709** [comment]
image

**Comment by Wilchenx** [comment]
> image

execute order 66

**Comment by DatBoiTim** [comment]
Perfectly balanced... as all things should be.

**Pull request merged**